types/pouchdb-core/index.d.ts

```diff
@@ -16,8 +16,10 @@ interface Blob {
     slice(start?: number, end?: number, contentType?: string): Blob;
 }
 
-interface Buffer extends Uint8Array {
-    write(string: string, offset?: number, length?: number, encoding?: string): number;
+interface Buffer {
```

Comment:
Can you bring back `extends Uint8Array`? 2 methods are missing to fix errors:
```ts
    reverse(): this;
    subarray(start?: number, end?: number): Buffer;
```

Also you can add these missing methods:
```ts
    writeBigInt64BE(value: bigint, offset?: number): number;
    writeBigInt64LE(value: bigint, offset?: number): number;
    writeBigUInt64BE(value: bigint, offset?: number): number;
    writeBigUInt64LE(value: bigint, offset?: number): number;
    readBigUInt64BE(offset?: number): bigint;
    readBigUInt64LE(offset?: number): bigint;
    readBigInt64BE(offset?: number): bigint;
    readBigInt64LE(offset?: number): bigint;
```

Reply:
@sandersn `extends Uint8Array` is still needed to have all `Buffer` methods :)

Reply:
No, the problem is that all `extends` clauses must be identical and the original one is `interface Buffer extends BufferModuleInteral {`. Inlining them as you suggest is the only way to get those methods.

Reply:
No, as far I know `interfaces` are merged by comparing properties, not by extension chain.

I've tested this PR's last commit and it's working fine with `extends Uint8Array` and `node`.

Reply:
> Inlining them as you suggest is the only way to get those methods.

`reverse` and `subarray` are redefined by `Buffer`, thus was needed to fix incompatible definitions when using `Uint8Array`.

`BigInt` were just missing in `pouchdb` definition.